webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -855,8 +905,24 @@ trait PromiseGeneration_BuildNodeContext {
           nodeTargets  =  allGroups.getTarget(info).map(_._2).toList
           timeMerge    =  System.nanoTime
           mergedProps  <- MergeNodeProperties.forNode(info, nodeTargets, nodeParam.map{case(k,v) => (k, v)}.toMap).toBox
+          nodeContextBefore  <- systemVarService.getSystemVariables(info, allNodeInfos, nodeTargets, globalSystemVariables, globalAgentRun, globalComplianceMode: ComplianceMode)
+          // Not sure if I should InterpolationContext or create a "EngineInterpolationContext
+          contextEngine = InterpolationContext(
+              info
+            , policyServer
+            , globalPolicyMode
+            , nodeContextBefore
+            , nodeParam.map{case(k, g) => (k, g.value)}.toMap
+          )
           _            =  {timeNanoMergeProp = timeNanoMergeProp + System.nanoTime - timeMerge}
-          nodeInfo     =  info.modify(_.node.properties).setTo(mergedProps.map(_.prop))
+          propsCompiled = mergedProps.map { p =>
+                            p.copy(prop = NodeProperty(p.prop.config.getString("name"), parseJValue(p.prop.toJson, contextEngine).toString.toConfigValue, None, None))
+                          }
+          _ <- effectUioUnit {mergedProps.foreach(p => println(p.prop.config))}.toBox
+          _ <- effectUioUnit(
+            println(s"======== JSON INTERPOLLATION : ${parseJValue(mergedProps(2).prop.toJson.\("value"), contextEngine).toString}")
```

Comment:
that will need to be removed

Reply:
Btw if you want to log, either use a logger or do 

_ = println(youmessage)

You don't neet it to be in a zio